### PR TITLE
DAOS-10690 test: Rectify TestPool.pool_percentage_used method

### PR DESCRIPTION
**Pull request by saurabhtandan**
Correcting TestPool.pool_percentage_used to use 'used' space instead of 'free' space.
Also added a method TestPool.get_pool_used_space for future use.

Test-tag: ec_aggregation

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/test_utils_pool.py [hunk 1]
```diff
@@ -1006,8 +1026,10 @@ def pool_percentage_used(self):
         """
         daos_space = self.get_pool_daos_space()
         pool_percent = {
-            'scm': round(float(daos_space["s_free"][0]) / float(daos_space["s_total"][0]) * 100, 4),
-            'nvme': round(float(daos_space["s_free"][1]) / float(daos_space["s_total"][1]) * 100, 4)
+            'scm': round(float(daos_space["s_total"][0] - daos_space["s_free"][0]) /
+                         float(daos_space["s_total"][0]) * 100, 4),
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_pool.py:1030]: `W504 line break after binary operator`
We should result these. I think...
```suggestion
            'scm': round(float(daos_space["s_total"][0] - daos_space["s_free"][0])
                         / float(daos_space["s_total"][0]) * 100, 4),
```

File: src/tests/ftest/util/test_utils_pool.py [hunk 2]
```diff
@@ -1006,8 +1026,10 @@ def pool_percentage_used(self):
...
+            'nvme': round(float(daos_space["s_total"][1] - daos_space["s_free"][1]) /
+                          float(daos_space["s_total"][1]) * 100, 4)
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_pool.py:1032]:
```suggestion
            'nvme': round(float(daos_space["s_total"][1] - daos_space["s_free"][1])
                          / float(daos_space["s_total"][1]) * 100, 4)
```

File: src/tests/ftest/util/test_utils_pool.py [hunk 3]
```diff
@@ -960,6 +960,26 @@ def get_pool_free_space(self, device="scm"):
             free_space = daos_space["s_free"][1]
         return free_space
 
+    def get_pool_used_space(self, device="scm"):
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_pool.py:963]: Question - why add this if it isn't being used yet?

**Review by daltonbohning — CHANGES_REQUESTED** [review]
-1 for the introduced flake warnings

File: src/tests/ftest/util/test_utils_pool.py (hunk 3 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/test_utils_pool.py:963]: As mentioned in the commit message, it is for future use. I thought it is something that can come in handy. Don't see any harm in adding something like this.

Comment (daltonbohning) [reply src/tests/ftest/util/test_utils_pool.py:963]: It's just extra code to maintain that isn't being used. If something changes to break the function, we won't know until someone tries to use the function

Comment (saurabhtandan) [reply src/tests/ftest/util/test_utils_pool.py]: It's removed now. IMO having something like get_pool_used_space() where we already have  get_pool_free_space() could have been useful at some point.

Comment (daltonbohning) [reply src/tests/ftest/util/test_utils_pool.py]: I agree it could be useful, but I think it's better to add when needed. Thanks!

**Review by daltonbohning — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Pull request merged**